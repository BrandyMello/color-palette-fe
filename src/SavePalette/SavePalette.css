.add_palette_form {
  display: flex;
  height: 75px;
  width: 600px;
  margin-left: 31%;
}

.palette_input {
  display: inline-block;
  width: 450px;
  height: 48px;
}

.add_palette_button {
  display: inline-block;
  width: 144px;
  height: 54px;
}