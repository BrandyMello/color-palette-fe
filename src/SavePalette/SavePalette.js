import React, { Component } from "react";
import "./SavePalette.css";
import ProjectsDropDown from "../ProjectsDropDown/ProjectsDropDown";

class SavePalette extends Component {
  constructor() {
    super();
    this.state = {
      project: "",
      paletteName: "",
      palettes: []
    };
  }

  handleChange = e => {
    this.setState({ [e.target.name]: e.target.value });
  };

  handleDropDown = project => {
    this.setState({ project });
  };

  getProjectId = () => {
    const matchingProjectId = this.props.projects.find(project => {
      if (project.name === this.state.project) {
        return project.id;
      }
    });
    console.log(matchingProjectId)
    return matchingProjectId.id;
  };

  render() {
    return (
      <div>
        <ProjectsDropDown
          projects={this.props.projects}
          handleDropDown={this.handleDropDown}
        />
        <form className="add_palette_form">
          <input
            className="palette_input"
            type="text"
            id="name"
            name="paletteName"
            placeholder="New Palette Name"
            value={this.state.paletteName}
            onChange={this.handleChange}
          />
          <button
            className="add_palette_button"
            name="newPalette"
            type="submit"
            onClick={() =>
              this.props.saveNewPaletteToProject({
                name: this.state.paletteName,
                projectName: this.state.project,
                colorOne: this.props.colors[0].color_1,
                colorTwo: this.props.colors[1].color_2,
                colorThree: this.props.colors[2].color_3,
                colorFour: this.props.colors[3].color_4,
                colorFive: this.props.colors[4].color_5,
                projectId: this.getProjectId()
              })
            }
          >
            Save Palette
          </button>
        </form>
      </div>
    );
  }
}

export default SavePalette;
