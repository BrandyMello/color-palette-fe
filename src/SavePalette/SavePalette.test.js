import React from "react";
import { shallow } from "enzyme";
import SavePalette from "./SavePalette";

describe("SavePalette", () => {
  let wrapper;
  let mockPalette;
  let mockProjects;

  beforeEach(() => {
    mockProjects = [{ name: "SeaBreeze", id: 1 }];

    mockPalette = [
      {
        name: "SeaBreeze",
        projectName: "Cool Colors",
        colorOne: "#013163",
        colorTwo: "#015E7E",
        colorThree: "#00AAB2",
        colorFour: "#C5BBCC",
        colorFive: "#A96B70",
        projectId: 1
      }
    ];

    wrapper = shallow(
      <SavePalette
        projects={mockProjects}
        colors={mockPalette}
        saveNewPaletteToProject={jest.fn()}
      />
    );
  });

  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should update state when handleChange is called", () => {
    const mockEvent = {
      target: {
        name: "paletteName",
        value: {
          name: "SeaBreeze",
          projectName: "Cool Colors",
          colorOne: "#013163",
          colorTwo: "#015E7E",
          colorThree: "#00AAB2",
          colorFour: "#C5BBCC",
          colorFive: "#A96B70",
          projectId: 1
        }
      }
    };
    const expected = {
      name: "SeaBreeze",
      projectName: "Cool Colors",
      colorOne: "#013163",
      colorTwo: "#015E7E",
      colorThree: "#00AAB2",
      colorFour: "#C5BBCC",
      colorFive: "#A96B70",
      projectId: 1
    };

    wrapper.instance().handleChange(mockEvent);

    expect(wrapper.state("paletteName")).toEqual(expected);
  });
});
