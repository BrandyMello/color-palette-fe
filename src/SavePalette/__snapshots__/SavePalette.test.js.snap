// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SavePalette should match snapshot 1`] = `
<form
  className="add_palette_form"
>
  <ProjectsDropDown
    handleDropDown={[Function]}
    projects={
      Array [
        Object {
          "id": 1,
          "name": "SeaBreeze",
        },
      ]
    }
  />
  <input
    className="palette_input"
    id="name"
    name="paletteName"
    onChange={[Function]}
    placeholder="New Palette Name"
    type="text"
    value=""
  />
  <button
    className="add_palette_button"
    name="newPalette"
    onClick={[Function]}
    type="submit"
  >
    Save Palette
  </button>
</form>
`;
